lib/ood_core/job/adapters/linux_systemd/launcher.rb

```diff
@@ -0,0 +1,247 @@
+    args  = args.map(&:to_s)
+    env = env.to_h
+    o, e, s = Open3.capture3(env, cmd, *args, stdin_data: stdin.to_s)
+    s.success? ? o : raise(Error, e)
+  end
+
+  # The full command to ssh into the destination host and execute the command.
+  # SSH options include:
+  # -t Force pseudo-terminal allocation (required to allow tmux to run)
+  # -o BatchMode=yes (set mode to be non-interactive)
+  # if ! strict_host_checking
+  # -o UserKnownHostsFile=/dev/null (do not update the user's known hosts file)
+  # -o StrictHostKeyChecking=no (do no check the user's known hosts file)
+  #
+  # @param destination_host [#to_s] the destination host you wish to ssh into
+  # @param cmd [Array<#to_s>] the command to be executed on the destination host
+  def ssh_cmd(destination_host, cmd)
+    if strict_host_checking
+      [
+        'ssh', '-t',
+        '-o', 'BatchMode=yes',
+        "#{username}@#{destination_host}"
+      ].concat(cmd)
+    else
+      [
+        'ssh', '-t',
+        '-o', 'BatchMode=yes',
+        '-o', 'UserKnownHostsFile=/dev/null',
+        '-o', 'StrictHostKeyChecking=no',
+        "#{username}@#{destination_host}"
+      ].concat(cmd)
+    end
+  end
+
+  def shell
+    ENV['SHELL'] || '/bin/bash'
+  end
+
+  # Wraps a user-provided script into a systemd-run transient service
+  def wrapped_script(script, session_name)
+    content = script.content
+    unless user_script_has_shebang?(script)
+     content = "#!#{shell}\n#{content}"
+    end
+
+    ERB.new(
+      File.read(Pathname.new(__dir__).join('templates/script_wrapper.erb.sh'))
+    ).result(binding.tap {|bnd|
+      {
+        'arguments' => script_arguments(script),
+        'cd_to_workdir' => (script.workdir) ? "cd #{script.workdir}" : '',
+        'debug' => debug,
+        'email_on_terminated' => script_email_on_event(script, 'terminated'),
+        'email_on_start' => script_email_on_event(script, 'started'),
+        'environment' => export_env(script),
+        'error_path' => error_path(script),
+        'job_name' => script.job_name.to_s,
+        'output_path' => (script.output_path) ? script.output_path.to_s : '/dev/null',
+        'script_content' => content,
+        'script_timeout' => script_timeout(script),
+        'session_name' => session_name,
+        'ssh_hosts' => ssh_hosts,
+        'workdir' => (script.workdir) ? script.workdir.to_s : '/tmp',
+      }.each{
+        |key, value| bnd.local_variable_set(key, value)
+      }
+    })
+  end
+
+  # Generate the environment export block for this script
+  def export_env(script)
+    environment = script.job_environment
+    (environment ? environment : {}).map{
+      |key, value| "export #{key}=#{Shellwords.escape(value)}"
+    }.sort.join("\n")
+  end
+
+  def script_timeout(script)
+    wall_time = script.wall_time.to_i
+    return site_timeout if wall_time == 0
+    return [wall_time, site_timeout].min unless site_timeout == 0
+
+    wall_time
+  end
```

Comment:
I don't know about this one - it seems to me that the form.yml should not allow anything less than 1? I guess I'll take a look at how other adapters are doing it.

Reply:
Yea mostly I think it's a scheduler thing. If you don't give a walltime there's some default. I checked linux_host and `timeout 0` must mean it's forever?

Reply:
I think you are correct - I changed it so that if it ends up being 0 then it will set the limit to infinity - which is what systemd wants in that case. Thanks!